include/motor-control/core/brushed_motor/brushed_motion_controller.hpp

```diff
@@ -20,6 +25,36 @@ class MotionController {
     MotionController(MotionController&&) = delete;
     ~MotionController() = default;
 
+    void move(const can_messages::GripperGripRequest& can_msg) {
+        BrushedMove msg{.duration = can_msg.duration,
+                        .freq = can_msg.freq,
+                        .duty_cycle = can_msg.duty_cycle,
+                        .group_id = can_msg.group_id,
+                        .seq_id = can_msg.seq_id,
+                        .stop_condition = MoveStopCondition::none};
+        queue.try_write(msg);
+        hardware.grip();
```

Comment:
I thought we were going to be writing to a queue for handling the motor interrupts, so do we still need these `grip` and `home` functions?

Reply:
Essentially, the `grip` and `home` functions actually begin the PWM generation. Only after the PWM generation has started, the motor interrupt handler could try to read from the limit switch. 

I struggled to think of another better way to handle this, let me know what you think!

Reply:
Based on previous discussions - you're right, this is a problem. There's two solutions I can see:
1. Only turn the PWM channel off (so, CC1) when you're not running PWM; keep the timer core running, and generating overflow interrupts, all the time.
2. Use more of the hardware. The PWM pulse callback does two things: count down time remaining, and check the limit switch. The limit switch can be set up as a gpio external interrupt that sends a message to this task. The time remaining could be loaded in a timer compare register of its own. Then you wouldn't have any logic running in interrupts - you'd stop the PWM if you got an EXTI or if you got the timer duration countdown ping.

I think these all make sense; when we spoke earlier, I recommended 2, but given that at some point we're going to add a full position controller to this logic, 1 probably makes more sense at second thought.